llvm/include/llvm/ADT/APFloat.h

```diff
@@ -1127,19 +1129,11 @@ class APFloat : public APFloatBase {
   /// \param Semantics - type float semantics
   static APFloat getAllOnesValue(const fltSemantics &Semantics);
 
-  /// Returns true if the given semantics supports either NaN or Infinity.
+  /// Returns true if the given semantics supports NaN or Infinity.
   ///
   /// \param Sem - type float semantics
   static bool hasNanOrInf(const fltSemantics &Sem) {
-    switch (SemanticsToEnum(Sem)) {
-    default:
-      return true;
-    // Below Semantics do not support {NaN or Inf}
-    case APFloat::S_Float6E3M2FN:
-    case APFloat::S_Float6E2M3FN:
-    case APFloat::S_Float4E2M1FN:
-      return false;
-    }
+    return semanticsHasNan(Sem) || semanticsHasInf(Sem);
```

Comment:
Do we have unit tests that cover this already?

Reply:
`hasNanOrInf` is used in `APFloatTest.cpp`. (The test is a bit suspicious though. It calls `APFloat::getInf` **and** `APFloat::getNaN` if `hasNanOrInf` returns "true".

Reply:
Maybe worth adding something new based on `fltSemantics` then?

Reply:
We discussed offline. You can always ask a floating-point type that supports NaN but not Inf for an Inf. And what you get is a NaN. I added a comment to the test case to clarify that.